integration-tests/mssql_test.go

```diff
@@ -65,46 +85,86 @@ var _ = Describe("MSSQL", Label("MSSQL"), func() {
+				// Combine required properties and received params
+				// In case of collision params take precedence
+				combinedProperties := map[string]any{}
+				maps.Copy(combinedProperties, requiredProperties)
+				maps.Copy(combinedProperties, defaultProperties)
+				maps.Copy(combinedProperties, params)
+				// If we included required properties directly in the plan we wouldn't be able to test
+				// certain scenarios easily. For example, passing an invalid engine value would require
+				// another test and another plan without the engine (user-inputs can't override plan properties)
+				//
+				// However, with this "COMPOSITIONAL" approach we have more fine-grained control over the inputs
+
+				_, err := broker.Provision(msSQLServiceName, "custom-sample", combinedProperties)
+
+				Expect(err).To(MatchError(expectedErrorMsg))
 			},
 			Entry(
 				"invalid region",
 				map[string]any{"region": "-Asia-northeast1"},
-				"region: Does not match pattern '^[a-z][a-z0-9-]+$'",
+				`unexpected status code 500: {"description":"1 error(s) occurred: region: Does not match pattern '^[a-z][a-z0-9-]+$'"}`+"\n",
 			),
 			Entry(
 				"instance name minimum length is 6 characters",
 				map[string]any{"instance_name": stringOfLen(5)},
-				"instance_name: String length must be greater than or equal to 6",
+				`unexpected status code 500: {"description":"1 error(s) occurred: instance_name: String length must be greater than or equal to 6"}`+"\n",
 			),
 			Entry(
 				"instance name maximum length is 98 characters",
 				map[string]any{"instance_name": stringOfLen(99)},
-				"instance_name: String length must be less than or equal to 98",
+				`unexpected status code 500: {"description":"1 error(s) occurred: instance_name: String length must be less than or equal to 98"}`+"\n",
 			),
 			Entry(
 				"instance name invalid characters",
 				map[string]any{"instance_name": ".aaaaa"},
-				"instance_name: Does not match pattern '^[a-z][a-z0-9-]+$'",
+				`unexpected status code 500: {"description":"1 error(s) occurred: instance_name: Does not match pattern '^[a-z][a-z0-9-]+$'"}`+"\n",
 			),
 			Entry(
 				"database name maximum length is 64 characters",
 				map[string]any{"db_name": stringOfLen(65)},
-				"db_name: String length must be less than or equal to 64",
+				`unexpected status code 500: {"description":"1 error(s) occurred: db_name: String length must be less than or equal to 64"}`+"\n",
+			),
+			Entry(
+				"engine must be one of the allowed values",
+				map[string]any{"engine": "not-an-allowed-engine"},
+				`unexpected status code 500: {"description":"1 error(s) occurred: engine: engine must be one of the following: \"sqlserver-ee\", \"sqlserver-ex\", \"sqlserver-se\", \"sqlserver-web\""}`+"\n",
+			),
+			Entry(
+				"storage_gb minimum value is 20",
+				map[string]any{"storage_gb": 19},
+				`unexpected status code 500: {"description":"1 error(s) occurred: storage_gb: Must be greater than or equal to 20"}`+"\n",
 			),
```

Comment:
We can test we have flagged required values as expected.

Reply:
Added new tests. Lines from 96 to 106, although a "similar" test was already available in 86 to 94.